api/utxoCache.ts

```diff
@@ -0,0 +1,109 @@
+import { getAddressUtxoOrdinalBundles, getUtxoOrdinalBundle } from './ordinals';
+
+export type UtxoStates = 'inscribed' | 'notInscribed' | 'unknown';
+
+export type UtxoCacheStruct = {
+  [utxoId: string]: UtxoStates;
+};
+
+export type StorageAdapter = {
+  get(key: string): Promise<string | null> | string | null;
+  set(key: string, value: string): Promise<void> | void;
+  remove(key: string): Promise<void> | void;
+};
+
+type UtxoCacheConfig = {
+  cacheStorageController: StorageAdapter;
+};
+
+export class UtxoCache {
+  private readonly _cacheStorageController: StorageAdapter;
+
+  CACHE_STORAGE_KEY = 'utxoCache';
+
+  CACHE_VERSION_STORAGE_KEY = 'utxoCacheVersion';
```

Comment:
Could we rather tie the cache to the version directly to avoid an erroneous situation where we update the version and the app closes before the cache update is complete?

We could have a single storage key and the cache could look like:
```
type UtxoCache = {
  version: number;
  utxos: UtxoCacheStruct;
}
```